dietpi/dietpi-software

```diff
@@ -11935,6 +11942,56 @@ _EOF_
 			G_EXEC_OUTPUT=1 G_EXEC docker run -d -p '9002:9000' --name=portainer --restart=always -v '/run/docker.sock:/var/run/docker.sock' -v 'portainer_data:/data' 'portainer/portainer-ce'
 		fi
 
+		if To_Install 141 adsb-setup adsb-docker # ADS-B Feeder
+		then
+			# clone the adsb-feeder repo into /tmp
+			G_EXEC_OUTPUT=1 G_EXEC git clone -b dietpi 'https://github.com/dirkhh/adsb-feeder-image.git' /tmp/adsb-feeder
+
+			# remove the service that isn't needed for an app install on DietPi and install the rest
+			G_EXEC cd /tmp/adsb-feeder/src/modules/adsb-feeder/filesystem/root
+			G_EXEC rm ./usr/lib/systemd/system/adsb-bootstrap.service
+			G_EXEC mv ./usr/lib/systemd/system/* /etc/systemd/system/
+
+			# determine the version
+			local ADSB_FEEDER_DATE_COMPONENT=$(git log -20 --date='format:%y%m%d' --format='%ad' | uniq -c | mawk '{print $2"."$1;exit}')
+			local ADSB_FEEDER_TAG_COMPONENT=$(git describe --match 'v[0-9]*' | cut -d- -f1)
+			local ADSB_FEEDER_VERSION="$ADSB_FEEDER_TAG_COMPONENT(dietpi)-$ADSB_FEEDER_DATE_COMPONENT"
+
+			# create the target directory for the app and populated with the code from the git checkout
+			[[ -d '/opt/adsb' ]] && G_EXEC rm -R /opt/adsb
+			G_EXEC mv /tmp/adsb-feeder/src/modules/adsb-feeder/filesystem/root/opt/adsb /opt/
+			G_EXEC cd /opt/adsb
+
+			# remove the git clone of the repo we installed from
+			G_EXEC rm -R /tmp/adsb-feeder
+
+			# create a symlink so the config files reside where they should be in /mnt/dietpi_userdata/adsb-feeder
+			G_EXEC mkdir -p /mnt/dietpi_userdata/adsb-feeder/config
+			G_EXEC ln -s /mnt/dietpi_userdata/adsb-feeder/config .
+
+			# set the 'image name' and version that are shown in the footer of the Web UI
+			G_EXEC eval 'echo '\''ADSB Feeder app running on DietPi'\'' > feeder-image.name'
+			G_EXEC eval "echo '$ADSB_FEEDER_VERSION' > adsb.im.version"
+
+			# get the Python module
+			# for older distros we need to install via pip in order to get flask 2
+			if (( $G_DISTRO < 7 ))
+			then
+				G_EXEC_OUTPUT=1 G_EXEC pip3 install -U flask
+			else
+				G_AGI python3-flask
+			fi
+
+			# finally ensure that /run allows executables (this is needed for the containers to be able to not do excessive
+			# writes to physical storage - which might be an SD card)
+			G_EXEC mount -o remount,exec /run
```

Comment:
Last question: This is really only needed once, for `adsb-nonimage.service` or so? Just because it is not boot persistent and also intended to not be so. Actually using a different tmpfs might be even better to keep `/run` non-executable at all time.

Reply:
I'm on a plane right now, tracking my own flight using the ADS-B Feeder on DietPi on an RPi3a.
Definitely getting a fair amount of testing.

The `/run` executable is always needed. And after a reboot the service manages it if necessary.
If you would prefer a different tempfs I can of course switch, but there is really no security benefit from having `/run` `noexec` (and most distros don't do that - DietPi is one of the few that does)

Reply:
The `noexec` is not coming from DietPi, but from Debian, or the responsible Debian package. We do not touch its mount options. But I'm also not sure whether this is a real security benefit, respectively whether anything is relying on it and would cause a security issue with `noexec` removed.

However, it is okay for me to check back about this later and merge this PR. Wishing a pleasant flight, and again, many thanks for this great implementation 🙂!